### PR TITLE
Support loading images from urls without extensions

### LINKED ISSUE #162: altair_tiles: Tiles are missing in PNG export but works with SVG
In https://github.com/vega/altair_tiles/issues/38 it came up that for certain tile providers, the tiles are missing in an exported PNG but they show up in a Jupyter notebook and in an SVG chart.

# Minimal example

```python
import altair as alt
import altair_tiles as til
import geopandas as gpd

gdf_ne = gpd.read_file("https://naciscdn.org/naturalearth/110m/cultural/ne_110m_admin_0_countries.zip")  # zipped shapefile
extent_roi = gdf_ne.loc[gdf_ne["CONTINENT"] == "Africa", ["CONTINENT", "geometry"]]

chart = (
    alt.Chart(extent_roi)
    .mark_geoshape(fillOpacity=0.1, stroke="green", strokeWidth=2)
    .project(type="mercator")
)
chart = til.add_tiles(chart, provider=til.providers.Esri.WorldGrayCanvas).properties(
    width=600, height=400
)
chart
```
Chart displays as expected with tiles:

image

When saving to SVG and PNG, only the SVG image shows the tiles so seems like they get lost when downloading and embedding them.
```python
chart.save("chart.svg")
chart.save("chart.png")
```
If you switch the provider in the above example to `til.providers.OpenStreetMap.Mapnik`, the tiles also show up in PNG. Sorry I don't know how to debug this any further but maybe it's something related to #63 where certain providers require specific request headers or so? In #64 you mentioned that we could set `RUST_LOG` to `vl_convert` but at least in a notebook, nothing shows up and I don't have cargo set up to install the CLI.

### PR DESCRIPTION
**Pull request by jonmmease**
Closes #162 

The issue was that this tile provider serves images in jpg format, but the URLs do not include a file extension (e.g. https://server.arcgisonline.com/ArcGIS/rest/services/Canvas/World_Light_Gray_Base/MapServer/tile/5/14/17) and usvg uses the file extension to infer the image type.  This PR adds logic to infer the image type from the `Content-Type` header and use that to build a file extension that usvg is happy with.

cc @binste

### CODE REVIEW

File: DEVELOP.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ Updating the CI build to use QEMU seems to have fixed the issue. Leaving this co
  > The Deno dependencies currently correspond to Deno 1.30.3. When updating to later versions, we've run into errors for packages cross compiled to Linux aarch64. See https://github.com/vega/vl-convert/issues/52. Be sure to test this scenario when updating Deno in the future.
 
 ## Debug Logging
-To enable logging, set the RUST_LOG environment variable to "vl_convert"
+To enable logging, set the RUST_LOG environment variable to info, warn, or error
 ```
-RUST_LOG=vl_convert
+RUST_LOG=info
 ```
```
Comment (jonmmease) [inline DEVELOP.md:63]: This way logging info for usvg is displayed as well

File: thirdparty_rust.yaml
Comment (jonmmease) [inline thirdparty_rust.yaml:1]: update to cargo-bundle-license crate

File: vl-convert-pdf/src/lib.rs [hunk 2]
```diff
@@ -424,10 +424,7 @@ fn node_has_zero_opacity(node: &Node) -> bool {
 }
 
 fn get_text_width(text: &Text, font_db: &Database) -> Option<f64> {
-    let Some(node) = text.convert(font_db, Default::default()) else {
-        return None;
-    };
-    get_text_width_from_path(node)
+    get_text_width_from_path(text.convert(font_db, Default::default())?)
```
Comment (jonmmease) [inline vl-convert-pdf/src/lib.rs:427]: clippy fix

File: vl-convert-rs/src/image_loading.rs [hunk 3]
```diff
@@ -29,21 +29,18 @@ pub type ImageHrefStringResolverFn = Box<dyn Fn(&str, &Options) -> Option<ImageK
 pub fn custom_string_resolver() -> ImageHrefStringResolverFn {
     let default_string_resolver = ImageHrefResolver::default_string_resolver();
     Box::new(move |href: &str, opts: &Options| {
+        info!("Resolving image: {href}");
```
Comment (jonmmease) [inline vl-convert-rs/src/image_loading.rs:32]: added logging

**Comment by mattijn** [comment]
Very clever solution! Really cool to read your code🙌

**Pull request merged**